lib/crypto.js

```diff
@@ -584,17 +584,22 @@ ECDH.prototype.generateKeys = function generateKeys(encoding, format) {
 ECDH.prototype.getPublicKey = function getPublicKey(encoding, format) {
   var f;
   if (format) {
-    if (typeof format === 'number')
-      f = format;
-    if (format === 'compressed')
-      f = constants.POINT_CONVERSION_COMPRESSED;
-    else if (format === 'hybrid')
-      f = constants.POINT_CONVERSION_HYBRID;
-    // Default
-    else if (format === 'uncompressed')
-      f = constants.POINT_CONVERSION_UNCOMPRESSED;
-    else
-      throw new TypeError('Bad format: ' + format);
+    switch (format.length) {
+      case 6:
+        if (format === 'hybrid')
+          f = constants.POINT_CONVERSION_HYBRID;
+        break;
+      case 10:
+        if (format === 'compressed')
+          f = constants.POINT_CONVERSION_COMPRESSED;
+        break;
+      case 12:
+        if (format === 'uncompressed')
+          f = constants.POINT_CONVERSION_UNCOMPRESSED;
+        break;
+      default:
+        throw new TypeError('Bad format: ' + format);
```

Comment:
This is true but the error should also be thrown in case the `format.length` will match one of the entries without matching the correct word. That is not the case anymore.